modules/nf-core/kraken2/build/main.nf

```diff
@@ -1,36 +1,39 @@
 process KRAKEN2_BUILD {
-    tag "$meta.id"
+    tag "${meta.id}"
     label 'process_medium'
     conda "${moduleDir}/environment.yml"
-    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/0f/0f827dcea51be6b5c32255167caa2dfb65607caecdc8b067abd6b71c267e2e82/data' :
-        'community.wave.seqera.io/library/kraken2_coreutils_pigz:920ecc6b96e2ba71' }"
+    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
+        ? 'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/0f/0f827dcea51be6b5c32255167caa2dfb65607caecdc8b067abd6b71c267e2e82/data'
+        : 'community.wave.seqera.io/library/kraken2_coreutils_pigz:920ecc6b96e2ba71'}"
 
     input:
-    tuple val(meta), path(db)
+    tuple val(meta), path(library_added_files, stageAs: "database/library/added/")
+    tuple val(meta2), path(seqid2taxid_map, stageAs: "database/seqid2taxid.map")
+    tuple val(meta3), path(taxonomy_files, stageAs: "database/taxonomy/")
     val cleaning
 
     output:
-    tuple val(meta), path("$prefix"), emit: db
-    path "versions.yml"             , emit: versions
+    tuple val(meta), path("${prefix}"), emit: db
+    path "versions.yml", emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
     prefix = task.ext.prefix ?: "${meta.id}"
-    runclean = cleaning ? "kraken2-build --clean --db ${db}" : ""
+    runclean = cleaning ? "kraken2-build --clean --db database/" : ""
     """
     kraken2-build \\
         --build \\
-        $args \\
+        ${args} \\
         --threads ${task.cpus} \\
-        --db ${db}
-    $runclean
-    if [[ \$(basename ${db}) != "${prefix}" ]]; then
-        mv ${db}/* ${prefix}
-    fi
+        --db database/
+
+    ${runclean}
+
+    mv database/ ${prefix}/
```

Comment:
This will move `database` inside `prefix`, e.g. `${prefix}/database/hash.k2d` which is not exactly what you have in the stub section (and in the previous implementation): `touch ${prefix}/{hash,opts,tax}.k2d`

Reply:
Yes, indeed